Source Code/Ticket Management System/supportgiver_menu.c

```diff
@@ -77,30 +104,74 @@ void browseTickets()
 					}
 					case('2'): // Filter by tier
 					{
-						// TODO
+						wchar_t* cty;
+						wchar_t tier[32];
+						wprintf(L"%s\n", L"please input tier");
+						wscanf(L"%s", tier);
+						wcstok(tier, "\n", &cty);
+						struct Vector* filtered_vector = vector_filter(tickets, filterByTier, tier);
+						printTicketsArray(filtered_vector);
+						vector_destroy(filtered_vector);
 						break;
 					}
 					case('3'): // Filter by type
 					{
-						// TODO
+						wchar_t* ctz;
+						wchar_t type[32];
+						wprintf(L"%s\n", L"please input type");
+						wscanf(L"%s", type);
+						wcstok(type, "\n", &ctz);
+						struct Vector* filtered_vector = vector_filter(tickets, filterByType, type);
+						printTicketsArray(filtered_vector);
+						vector_destroy(filtered_vector);
 						break;
 					}
-					case('4'): // Filter by stakeholders
+					case('4'): // sort by severity
 					{
-						// TODO
+
+						struct Vector* sorted_vector = vector_sort(tickets, sortBySeverity);
+						printTicketsArray(sorted_vector);
+						vector_destroy(sorted_vector);
+						break;
+					}
+					case('5'): // Filter by stakeholders
+					{
+						wchar_t* ct;
+						wchar_t stakeholders[32];
+						wprintf(L"%s\n", L"please input stakeholders");
+						wscanf(L"%s", stakeholders);
+						wcstok(stakeholders, "\n", &ct);
+						struct Vector* filtered_vector = vector_filter(tickets, filterByStakeholders, stakeholders);
+						printTicketsArray(filtered_vector);
+						vector_destroy(filtered_vector);
 						break;
 					}
-					case('5'): // Search by customer email
+					case('6'): // Search by customer email
 					{
-						// TODO
+						wchar_t* ce;
+						wchar_t customer_email[32];
+						wprintf(L"%s\n", L"please input Customer Email");
+						wscanf(L"%s", customer_email);
+						wcstok(customer_email, "\n", &ce);
+						struct Vector* filtered_vector = vector_filter(tickets, searchByCustomerEmail, customer_email);
+						printTicketsArray(filtered_vector);
+						vector_destroy(filtered_vector);
 						break;
 					}
-					case('6'): // Search by tags
+					case('7'): // Search by tags
 					{
-						// TODO
+						wchar_t* cta;
+						wchar_t tags[32];
+						wprintf(L"%s\n", L"please input Customer Email");
```

Comment:
Asking for customer email but actually inputting tags.

Reply:
Still an issue.

Reply:
Still...

Reply:
Still...